pkg/kubectl/resource_columns.go

```diff
@@ -0,0 +1,593 @@
+	{title: "SECRETS", content: func(s *api.ServiceAccount) string { return fmt.Sprintf("%d", len(s.Secrets)) }},
+	ageColumn,
+}
+
+var persistentVolumeColumns = []funcColumnPrinter{
+	noNamespaceColumn,
+	makeNameColumn("NAME"),
+	{title: "LABELS", content: func(p *api.PersistentVolume) string { return formatLabels(p.Labels) }},
+	{title: "CAPACITY", content: func(p *api.PersistentVolume) string {
+		aQty := p.Spec.Capacity[api.ResourceStorage]
+		aSize := aQty.String()
+		return aSize
+	}},
+	{title: "ACCESSMODES", content: func(p *api.PersistentVolume) string { return volume.GetAccessModesAsString(p.Spec.AccessModes) }},
+	{title: "STATUS", content: func(p *api.PersistentVolume) string { return fmt.Sprintf("%s", p.Status.Phase) }},
+	{title: "CLAIM", content: func(p *api.PersistentVolume) string {
+		claimRefUID := ""
+		if p.Spec.ClaimRef != nil {
+			claimRefUID += p.Spec.ClaimRef.Namespace
+			claimRefUID += "/"
+			claimRefUID += p.Spec.ClaimRef.Name
+		}
+		return claimRefUID
+	}},
+	{title: "REASON", content: func(p *api.PersistentVolume) string { return p.Status.Reason }},
+	ageColumn,
+}
+
+var persistentVolumeClaimColumns = []funcColumnPrinter{
+	namespaceColumn,
+	makeNameColumn("NAME"),
+	{title: "STATUS", content: func(p *api.PersistentVolumeClaim) string { return fmt.Sprintf("%s", p.Status.Phase) }},
+	{title: "LABELS", content: func(p *api.PersistentVolumeClaim) string { return formatLabels(p.Labels) }},
+	{title: "VOLUME", content: func(p *api.PersistentVolumeClaim) string { return p.Spec.VolumeName }},
+	{title: "CAPACITY", content: func(p *api.PersistentVolumeClaim) string {
+		if p.Spec.VolumeName != "" {
+			storage := p.Status.Capacity[api.ResourceStorage]
+			return storage.String()
+		}
+		return ""
+	}},
+	{title: "ACCESSMODES", content: func(p *api.PersistentVolumeClaim) string {
+		if p.Spec.VolumeName != "" {
+			return volume.GetAccessModesAsString(p.Status.AccessModes)
+		}
+		return ""
+	}},
+	ageColumn,
+}
+
+var componentStatus = []funcColumnPrinter{
+	noNamespaceColumn,
+	makeNameColumn("NAME"),
+	{title: "STATUS", content: func(cs *api.ComponentStatus) string {
+		status := "Unknown"
+		for _, condition := range cs.Conditions {
+			if condition.Type == api.ComponentHealthy {
+				if condition.Status == api.ConditionTrue {
+					status = "Healthy"
+				} else {
+					status = "Unhealthy"
+				}
+				break
+			}
+		}
+		return status
+	}},
+	{title: "MESSAGE", content: func(cs *api.ComponentStatus) string {
+		message := ""
+		for _, condition := range cs.Conditions {
+			if condition.Type == api.ComponentHealthy {
+				message = condition.Message
+				break
+			}
+		}
+		return message
+	}},
+	{title: "ERROR", content: func(cs *api.ComponentStatus) string {
+		error := ""
+		for _, condition := range cs.Conditions {
+			if condition.Type == api.ComponentHealthy {
+				error = condition.Error
+				break
+			}
+		}
+		return error
+	}},
+}
+
+var columnMap = map[reflect.Type][]funcColumnPrinter{
```

Comment:
This map should be a map between `map[reflect.Type]ObjectPrinter` where
```
type ObjectPrinter func(runtime.Object, PrintContext) []ColumnPrinter
```

That way it is possible for extenders to register their own types and `ObjectPrinters`.  You'd still be able to perform the determination like you do in `getColumnPrintersFor`, but it would encapsulate the knowledge in a more structured way.